### PR TITLE
Add lint option to Makefile and fix lint issues

### PR DESCRIPTION
**Pull request by Rulox**
### Proposed changes
* A new option for Makefile has been added to run `golangci-lint run` during local dev.
* Adapted the code to pass all the linter rules.

### Checklist
Before creating a PR, run through this checklist and mark each as complete.

- [X] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/master/CONTRIBUTING.md) doc
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have checked that all unit tests pass after adding my changes
- [X] I have updated necessary documentation
- [X] I have rebased my branch onto master
- [X] I will ensure my PR is targeting the master branch and pulling from my branch from my own fork

### CODE REVIEW

**Comment by boxer3379** [comment]
323

File: cmd/nginx-ingress/main.go [hunk 1]
```diff
@@ -107,7 +107,12 @@ The external address of the service is used when reporting the status of Ingress
 
 func main() {
 	flag.Parse()
-	flag.Lookup("logtostderr").Value.Set("true")
+
+	err := flag.Lookup("logtostderr").Value.Set("true")
+
+	if err != nil {
+		fmt.Printf("Error setting logtostderr to true: %v", err)
```
Comment (pleshakov) [inline cmd/nginx-ingress/main.go]: better to fatalf here

File: internal/k8s/controller_test.go [hunk 2]
```diff
@@ -159,9 +159,20 @@ func TestIsNginxIngress(t *testing.T) {
 func TestCreateMergableIngresses(t *testing.T) {
 	cafeMaster, coffeeMinion, teaMinion, lbc := getMergableDefaults()
 
-	lbc.ingressLister.Add(&cafeMaster)
-	lbc.ingressLister.Add(&coffeeMinion)
-	lbc.ingressLister.Add(&teaMinion)
+	err := lbc.ingressLister.Add(&cafeMaster)
+	if err != nil {
+		t.Errorf("Error adding Ingress to the ingress lister: %v", err)
```
Comment (pleshakov) [inline internal/k8s/controller_test.go]: can we include the name of the Ingress here and in other places - `"Error adding Ingress %v to the ingress lister: %v", cafeMaster.Name, err` ?

File: internal/k8s/controller_test.go [hunk 3]
```diff
@@ -964,8 +1016,15 @@ func TestFindIngressesForSecret(t *testing.T) {
...
-			lbc.ingressLister.Add(&test.ingress)
-			lbc.secretLister.Add(&test.secret)
+			err = lbc.ingressLister.Add(&test.ingress)
+			if err != nil {
+				t.Errorf("Error adding Ingress to the ingress lister: %v", err)
+			}
+
+			err = lbc.secretLister.Add(&test.secret)
+			if err != nil {
+				t.Errorf("Error adding Ingress to the ingress lister: %v", err)
```
Comment (pleshakov) [inline internal/k8s/controller_test.go]: not an Ingress

File: internal/k8s/controller_test.go [hunk 4]
```diff
@@ -1152,9 +1211,20 @@ func TestFindIngressesForSecretWithMinions(t *testing.T) {
...
+			}
+
+			err = lbc.ingressLister.Add(&test.ingress)
+			if err != nil {
+				t.Errorf("Error adding Ingress to the ingress lister: %v", err)
+			}
+
+			err = lbc.secretLister.Add(&test.secret)
+			if err != nil {
+				t.Errorf("Error adding Ingress to the ingress lister: %v", err)
```
Comment (pleshakov) [inline internal/k8s/controller_test.go]: not an Ingress

File: internal/metrics/prometheus.go [hunk 5]
```diff
@@ -39,13 +39,17 @@ func runServer(port string, registry prometheus.Gatherer) {
...
 	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
-		w.Write([]byte(`<html>
+		_, err := w.Write([]byte(`<html>
 			<head><title>NGINX Ingress Controller</title></head>
 			<body>
 			<h1>NGINX Ingress Controller</h1>
 			<p><a href='/metrics'>Metrics</a></p>
 			</body>
 			</html>`))
+
```
Comment (pleshakov) [inline internal/metrics/prometheus.go]: usually we don't have any empty lines before handling the err

File: internal/k8s/controller_test.go (hunk 2 above)
Comment (Rulox) [reply internal/k8s/controller_test.go]: Ok

**Comment by Rulox** [comment]
Fixed @pleshakov

File: cmd/nginx-ingress/main.go [hunk 6]
```diff
@@ -106,8 +106,13 @@ The external address of the service is used when reporting the status of Ingress
 )
 
 func main() {
+
```
Comment (pleshakov) [inline cmd/nginx-ingress/main.go]: no need for a blank line

**Pull request merged**